src/Server.zig

```diff
@@ -732,6 +739,40 @@ fn handleConfiguration(server: *Server, json: std.json.Value) error{OutOfMemory}
     const arena = arena_allocator.allocator();
 
     var new_config: configuration.Configuration = .{};
+    const maybe_root_dir: ?[]const u8 = if (server.workspaces.items.len == 1) dir: {
+        const uri = std.Uri.parse(server.workspaces.items[0].uri) catch |err| {
+            log.err("failed to parse root uri for workspace {s}: {!}", .{
+                server.workspaces.items[0].uri, err,
+            });
+            break :dir null;
+        };
+        break :dir try uri.path.toRawMaybeAlloc(arena);
+    } else null;
+
+    check_relative: inline for (&.{
```

Comment:
This loop does nothing because `new_config` has not been initialized yet. This loop should come after the `inline for (fields, result) |field, json_value| {` loop further below which initializes the values in `new_config`.